tethys_cli/cli_helpers.py

```diff
@@ -57,3 +70,58 @@ def load_apps():
     sys.stdout = open(os.devnull, "w")
     django.setup()
     sys.stdout = stdout
+
+
+def generate_salt_string():
+    salt = bcrypt.gensalt()
+    return salt
+
+
+def gen_salt_string_for_setting(app_name, setting):
+    secret_yaml_file = TETHYS_HOME / "secrets.yml"
+    secret_settings = {}
+    with secret_yaml_file.open("r") as secret_yaml:
+        secret_settings = yaml.safe_load(secret_yaml) or {}
+        if app_name not in secret_settings["secrets"]:
+            write_warning(
+                f"No definition for the app {app_name} in the secrets.yml. Generating one..."
+            )
+            secret_settings["secrets"][app_name] = {}
+        if "custom_settings_salt_strings" not in secret_settings["secrets"][app_name]:
+            write_warning(
+                f"No custom_settings_salt_strings in the app {app_name} in the secrets.yml. Generating one..."
+            )
+            secret_settings["secrets"][app_name]["custom_settings_salt_strings"] = {}
+
+        last_salt_string = ""
+        signer = Signer()
+        if (
+            setting.name
+            in secret_settings["secrets"][app_name]["custom_settings_salt_strings"]
+        ):
+            last_salt_string = secret_settings["secrets"][app_name][
+                "custom_settings_salt_strings"
+            ][setting.name]
+            signer = Signer(salt=last_salt_string)
+        try:
+            secret_unsigned = signer.unsign_object(f"{setting.value}")
+        except BadSignature:
+            write_error(
+                f"The salt string for the setting {setting.name} in the app {app_name} can not be generated in the secrets.yml because, the salt string for the setting in the secrets.yml was changed/deleted"
+            )
+            write_warning(
+                "Please enter the secret custom settings in the application settings again"
+            )
+            return
```

Comment:
This is another place where a secrets.yml function could be used to keep code DRY.

Reply:
Lines 96 to 107 were refactored in the `gen_salt_string_for_setting` function to `secret_unsigned = setting.get_value()`. Calling the `get_value` function uses the function `secrets_signed_unsigned_value` function in the `/tethys_apps/utilties.py`